spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java

```diff
@@ -38,61 +36,48 @@ public class ProductDAO {
 	RestTemplate restTemplate;
 
 	@Autowired
-	private ProductService productService;
+	private FeignProductService feignRequest;
+
+	@Autowired
+	private RestTemplateProductService restTemplateRequest;
+
+	@Autowired
+	private WebClientProductService webClientRequest;
 
 	public ResultHolder<List<Product>> list() {
-		return productService.list();
+		return feignRequest.list();
 	}
 
 	public ResultHolder<Product> detail(String rId, String pId) {
-		return productService.detail(rId, pId);
+		return feignRequest.detail(rId, pId);
 	}
 
 	public ResultHolder<Product> detailHidden(String pId) {
-		return productService.detailHidden(pId);
+		return feignRequest.detailHidden(pId);
 	}
 
 	public ResultHolder<String> buy(String rId, String pId, Integer number) {
-		return productService.buy(RPCType.SpringCloud.name(), rId, pId, number);
+		return feignRequest.buy(RPCType.SpringCloud.name(), rId, pId, number);
 	}
 
 	public ResultHolder<List<Product>> listTemplate() {
-		if (restTemplate != null) {
-			return restTemplate.getForObject("http://product/list", ResultHolder.class);
-		}
-		return productService.list();
-	}
 
-	public ResultHolder<Product> detailTemplate(String rId, String pId) {
-		if (restTemplate != null) {
-			Map<String, String> params = new HashMap<>(2);
-			params.put("rId", rId);
-			params.put("pId", pId);
-			return restTemplate.getForObject("http://product/detail", ResultHolder.class,
-					params);
-		}
-		return productService.detail(rId, pId);
+		return restTemplateRequest.list();
 	}
 
-	@FeignClient(name = "product")
-	public interface ProductService {
+	public ResultHolder<Product> detailTemplate(String rId, String pId) {
```

Comment:
Called `detailByRestTemplate`? Please modify similar problems at the same time, such `detailByWebClient`.